src/core/meshcop/meshcop_tlvs.cpp

```diff
@@ -128,33 +164,137 @@ const ChannelMaskEntryBase *ChannelMaskBaseTlv::GetFirstEntry(void) const
     VerifyOrExit(GetLength() >= sizeof(ChannelMaskEntryBase));
 
     entry = reinterpret_cast<const ChannelMaskEntryBase *>(GetValue());
-    VerifyOrExit(GetLength() >= entry->GetSize(), entry = NULL);
+    VerifyOrExit(GetLength() >= entry->GetEntrySize(), entry = NULL);
 
 exit:
     return entry;
 }
 
-const ChannelMaskEntry *ChannelMaskBaseTlv::GetMaskEntry(uint8_t aChannelPage) const
+ChannelMaskEntryBase *ChannelMaskBaseTlv::GetFirstEntry(void)
+{
+    ChannelMaskEntryBase *entry = NULL;
+
+    VerifyOrExit(GetLength() >= sizeof(ChannelMaskEntryBase));
+
+    entry = reinterpret_cast<ChannelMaskEntryBase *>(GetValue());
+    VerifyOrExit(GetLength() >= entry->GetEntrySize(), entry = NULL);
+
+exit:
+    return entry;
+}
+
+otError ChannelMaskTlv::SetChannelMask(uint32_t aChannelMask)
+{
+    otError           error  = OT_ERROR_NONE;
+    uint8_t           length = 0;
+    ChannelMaskEntry *entry;
+
+    VerifyOrExit((entry = static_cast<ChannelMaskEntry *>(GetFirstEntry())) != NULL, error = OT_ERROR_NO_BUFS);
+
+#if OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT
+    if (aChannelMask & OT_RADIO_915MHZ_OQPSK_CHANNEL_MASK)
+    {
+        entry->Init();
+        entry->SetChannelPage(OT_RADIO_CHANNEL_PAGE_2);
+        entry->SetMask(aChannelMask & OT_RADIO_915MHZ_OQPSK_CHANNEL_MASK);
+
+        length += sizeof(MeshCoP::ChannelMaskEntry);
+
+        VerifyOrExit((entry = static_cast<MeshCoP::ChannelMaskEntry *>(entry->GetNext(this))) != NULL,
```

Comment:
if only `OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT` is defined, it may not be proper to always move `entry` forward and exit with `OT_ERROR_NO_BUFS`.